### DISCUSSION TITLE
Fluent Icons, Stream Geometries for use with PathIcon

### DISCUSSION THREAD

**Discussion by LaurentInSeattle** [AvaloniaUI/Avalonia#15759] [Q&A]
Hi all, 

The documentation and the music store tutorial both seem to indicate that you have to copy / paste the icon definitions from https://avaloniaui.github.io/icons.html and create a custom style xaml file to include in your app.

Is there a faster way ? 
Is there a file somewhere that already has all these paths ready to use with PathIcon? 

Thanks!

**Comment by stevemonaco** [comment]
IMO, it's silly bloat to include over a thousand icons in your app when you use very few. IIRC, libraries like MahApps,Metro.IconPacks shipped with something like 10MB+ of dependencies (including icon data) which is insane. You also need to be careful that any icon library uses lazy loading. You could create a `ResourceDictionary` for these with `StreamGeometry`, but all icons would load on startup (or maybe on the first icon fetch).

It adds about 20-30s per icon to pick it manually and include as source. I put everything in a C# file [like this](https://github.com/stevemonaco/ImageMagitek/blob/main/TileShop.UI/Resources/AppIcons.cs) and then use `x:Static` to reference in XAML.

**Comment by kyuranger** [comment]
I have a very simple way to do it, but you need some image processing skills (adobe illustrator).

Firstly, get the icon you want in https://fonts.google.com/icons?icon.query=us with SVG format.

Then using adobe illustrator to convert it into a single SVG path.

Using https://vecta.io/nano to compress the SVG file just generated by adobe illustrator and get the value of the SVG path that is compressed.

In Avalonia,
`<Path Data="SVG path" Fill="Black"></path>`

BTW, regardless of whether you use my method or not, I know that all SVG files in Avalonia should be compressed first through the website I mentioned above.
